src/TorchSharp/NN/Pooling/AdaptiveAvgPool2D.cs

```diff
@@ -53,6 +53,42 @@ static public AdaptiveAvgPool2d AdaptiveAvgPool2d(long[] outputSize)
                     }
                 }
             }
+
+            /// <summary>
+            /// Applies a 2D adaptive average pooling over an input signal composed of several input planes.
+            /// The output is of size H x W, for any input size.The number of output features is equal to the number of input planes.
+            /// </summary>
+            /// <param name="outputSize">The target output size (H,W) of the image of the form H x W.</param>
+            /// <returns></returns>
+            static public AdaptiveAvgPool2d AdaptiveAvgPool2d((long,long) outputSize)
+            {
+                long[] os = new long[] { outputSize.Item1, outputSize.Item2 };
```

Comment:
>long [](http://example.com/codeflow?start=16&length=4)

does this array be alive or referenced after calling THSNN_AdaptiveAvgPool2d_ctor? 

Just asking if we can have it a stack allocated array?

Reply:
It applies to a lot of APIs, some yet to be implemented. They are also on high-frequency paths, so if this is something that has a big performance impact, then let's look at that.